WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -97,9 +98,14 @@ def index
     else
       @competitions = Competition
     end
-    @competitions = @competitions.includes(:country).where(showAtAll: true).order_by_date
 
-    if @present_selected
+    if @by_announcement_selected
+      @competitions = @competitions.includes(:country).where(showAtAll: true).order_by_announcement_date.reverse_order
+    else
+      @competitions = @competitions.includes(:country).where(showAtAll: true).order_by_date
+    end
```

Comment:
Here you can probably just keep the common scope, and just add the order based on the parameter:
```suggestion
    @competitions = @competitions.includes(:country).where(showAtAll: true)
    @competitions = if @by_announcement_selected
                      @competitions.order_by_announcement_date
                    else
                      @competitions.order_by_date
                    end
```

Note: I intentionally removed the `reverse_order`, see my other comment.